transforms/ember-object/index.js

```diff
@@ -0,0 +1,197 @@
+module.exports = function transformer(file, api) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  // Helper functions
+  /**
+   * Transform instance properties to MemberExpressions
+   *
+   * For example: `prop: value` --> `this.prop = value`
+   *
+   * @param {*} instanceProps
+   * @returns Array [ ExpressionStatement ]
+   */
+  function instancePropsToExpressions(instanceProps) {
+    return instanceProps.map(instanceProp =>
+      j.expressionStatement(
+        j.assignmentExpression(
+          "=",
+          j.memberExpression(j.thisExpression(), instanceProp.key),
+          instanceProp.value
+        )
+      )
+    );
+  }
+
+  /**
+   * Transform functions to class methods
+   *
+   * For example { foo: function() { }} --> { foo() { }}
+   *
+   * @param {[]} functionProps
+   * @returns Array [MethodDefinition]
+   */
+  function functionPropsToMethods(functionProps = []) {
+    return functionProps.map(functionProp => {
+      return j.methodDefinition("method", functionProp.key, functionProp.value);
+    });
+  }
+
+  /**
+   * Create  a constructor method
+   *
+   * @param {[]} instanceProps
+   * @return Array [MethodDefinition]
+   */
+  function createConstructor(instanceProps = []) {
+    if (instanceProps.length) {
+      return [
+        j.methodDefinition(
+          "constructor",
+          j.identifier("constructor"),
+          j.functionExpression(
+            null,
+            [],
+            j.blockStatement(instancePropsToExpressions(instanceProps))
+          )
+        )
+      ]; // Todo: find a way to include params
```

Comment:
I don't think we should worry about including params for the time being. In general, params will force rewrites of classes, because constructors with EmberObject do not match up to constructors with just native classes. I do not believe this is codemoddable in a clean way.